### PR TITLE
Change Jenkins Docker Runner to CentOS7 so k-NN lib will not crash on OS with lower glibc version

### LINKED ISSUE #481: k-NN library x64 build should use generic march
In ODFE, we had [run into issues](https://github.com/opendistro-for-elasticsearch/k-NN/pull/164) with library incompatibilities due to the k-NN library being compiled on a machine with support for SIMD instruction extensions and being run on a machine without support for SIMD instruction extensions. We added a small fix in ODFE CD to resolve this. 

ARM is not impacted.

We should have a similar fix in OpenSearch.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Change Jenkins Docker Runner to CentOS7 so k-NN lib will not crash on OS with lower glibc version
 
### Issues Resolved
#481
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peterzhuamazon** [comment]
Hi @dblock @peternied this PR is to add the centos7 docker image, remove centos8 (as it will end its LTS by end of 2021), and run the build process in centos7 so that k-NN lib can be built correctly for x64 without crashing when load on CentOS7.

As a reminder we list CentOS7 as a support OS here: https://opensearch.org/docs/opensearch/install/compatibility/

Glibc Versions:
```
AL2 - 2.26
Ubuntu 20 - 2.27
Ubuntu 18 - 2.27
Ubuntu 16 - 2.23
Centos 7 - 2.17
Centos 8 - 2.28
RHEL 7 - 2.17
RHEL 8 - 2.28
```

**Review by peternied — COMMENTED** [review]
Why update the bundle workflow job image?  There were no build errors during the normal job.

File: bundle-workflow/Jenkinsfile [hunk 1]
```diff
@@ -33,7 +33,7 @@ pipeline {
                     agent {
                         docker {
                             label 'Jenkins-Agent-al2-x64-c54xlarge-Docker-Host'
-                            image 'opensearchstaging/ci-runner:al2-x64-arm64-jdk14-node10.24.1-20210924'
+                            image 'opensearchstaging/ci-runner:centos7-x64-arm64-jdk14-node10.24.1-20210924'
```
Comment (peternied) [inline bundle-workflow/Jenkinsfile:36]: It would be great if we could automate building these images, created #628

**Review by dblock — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
> > Why update the bundle workflow job image? There were no build errors during this job.
> 
> If I understand correctly the k-nn native code compiles but doesn't run?

knnlib will compile on AL2 but it will crash during runtime on CentOS7.

**Pull request merged**